deeplearning4j-nn/pom.xml

```diff
@@ -12,11 +12,13 @@
 
     <name>deeplearning4j-nn</name>
 
-    <properties>
-        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
-    </properties>
 
     <dependencies>
+        <dependency>
+            <groupId>org.deeplearning4j</groupId>
+            <artifactId>deeplearning4j-utility-iterators</artifactId>
+            <version>${project.version}</version>
```

Comment:
Do we want deeplearning4j-nn relying on deeplearning4j-utility-iterators?
My first thought here would be no - i.e., ```-nn``` should be very minimal.
That's why I was suggesting we keep the async iterators in -nn but move everything else - I *think* the async ones are the only ones that are used directly in -nn. If we move them to -nn, we can drop this dependency here?
Not sure how strict we want to be with the "-nn should be minimal" design...

Reply:
The reason we do this is because the neural net api itself requires the async iterators and the like.